network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java

```diff
@@ -407,34 +407,74 @@ public void deleteNetwork(UUID uuid, int variantNum) {
         }
     }
 
-    @SuppressWarnings("javasecurity:S5145")
-    public void cloneNetwork(UUID uuid, int sourceVariantNum, int targetVariantNum, String targetVariantId) {
+    public void cloneNetwork(UUID targetNetworkUuid, UUID sourceNetworkUuid, List<String> targetVariantIds) {
+        List<VariantInfos> networkVariantsInfo = getVariantsInfos(sourceNetworkUuid).stream()
+                .filter(v -> targetVariantIds.contains(v.getId()))
+                .sorted(Comparator.comparing(VariantInfos::getNum))
+                .collect(Collectors.toList());
+
+        targetVariantIds.stream()
+                .filter(element -> !networkVariantsInfo.stream()
```

Comment:
This could not scale well with a lot of variant. You can move creation of existing variant ids outside the stream and create a set instead of a list.